### DISCUSSION TITLE
IPv6 tunnel remote access no traffic ,but IPv4 tunnel work fine

### DISCUSSION THREAD

**Discussion by ghost** [strongswan/strongswan#694] [Q&A] [answered]
IPv6 in IPv4 tunnel work fine,all client windows, linux, android no problem.
IPv6 tunnel, all client can connect ,but all  web sites can not be reached(ping can not be reached also) for windows linux client,only android strongswan client can access part web sites (test-ipv6.com show IPv6 address normal over IPSec VPN,speedtest app test speed also normal,google.com can not be reached ).
As a comparison Wireguard,OpenVPN  work fine for IPv4 and IPv6(include IPv6 in IPv4 tunnel,IPv4 in IPv6 tunnel) on this server.
The following is my configuration, I can't find where the problem is.

server:GCP debian11

/etc/sysctl.conf
net.ipv4.ip_forward = 1
net.ipv4.conf.all.rp_filter = 0
net.ipv4.conf.default.rp_filter = 0
net.ipv4.conf.ens4.rp_filter = 0
net.ipv6.conf.all.forwarding = 1
net.ipv6.conf.ens4.accept_ra = 2

iptables -t nat -A POSTROUTING -o ens4 -j MASQUERADE
ip6tables -t nat -A POSTROUTING -o ens4 -j MASQUERADE
iptables -t mangle -A FORWARD -m policy --pol ipsec --dir in -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1361:1536 -j TCPMSS --set-mss 1360
iptables -t mangle -A FORWARD -m policy --pol ipsec --dir out -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1361:1536 -j TCPMSS --set-mss 1360
ip6tables -t mangle -A FORWARD -m policy --pol ipsec --dir in -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1361:1536 -j TCPMSS --set-mss 1360
ip6tables -t mangle -A FORWARD -m policy --pol ipsec --dir out -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1361:1536 -j TCPMSS --set-mss 1360

/etc/swanctl/swanctl.conf
connections {
    ikev2-pubkey {
        version = 2
        send_cert = always
        pools = primary-pool-ipv4, primary-pool-ipv6
        local-1 {
            auth = pubkey
            certs = serverCert.pem
            id = wower.cool
        }
        remote-1 {
            auth = pubkey
            id = %any
        }
        children {
            ikev2-pubkey {
                local_ts = 0.0.0.0/0,::/0
            }
        }
    }
}
pools {
    primary-pool-ipv4 {
        addrs = 10.20.0.0/16
        dns = 8.8.8.8, 8.8.4.4
    }
    primary-pool-ipv6 {
        addrs = fd00:bbbb:10:20::/64
        dns = 2001:4860:4860::8888, 2001:4860:4860::8844
    }
}
secrets {
    private-key {
        file = serverKey.pem
    }
}
authorities {
   strongswan {
      cacert = caCert.pem
      crl_uris = crl.pem
   }
}

swanctl --log
06[NET] received packet: from 2409:8a4c:cbe:4b50:a814:e811:3751:bf8a[58609] to 2600:1900:4120:c5b7:0:3::[500] (716 bytes)
06[ENC] parsed IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
06[IKE] 2409:8a4c:cbe:4b50:a814:e811:3751:bf8a is initiating an IKE_SA
06[CFG] selected proposal: IKE:AES_GCM_16_192/PRF_HMAC_SHA2_256/ECP_256
06[IKE] remote host is behind NAT
06[IKE] sending cert request for "CN=Root CA"
06[ENC] generating IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) CERTREQ N(FRAG_SUP) N(HASH_ALG) N(CHDLESS_SUP) N(MULT_AUTH) ]
06[NET] sending packet: from 2600:1900:4120:c5b7:0:3::[500] to 2409:8a4c:cbe:4b50:a814:e811:3751:bf8a[58609] (297 bytes)
06[NET] received packet: from 2409:8a4c:cbe:4b50:a814:e811:3751:bf8a[50995] to 2600:1900:4120:c5b7:0:3::[4500] (1348 bytes)
06[ENC] parsed IKE_AUTH request 1 [ EF(2/4) ]
06[ENC] received fragment #2 of 4, waiting for complete IKE message
10[NET] received packet: from 2409:8a4c:cbe:4b50:a814:e811:3751:bf8a[50995] to 2600:1900:4120:c5b7:0:3::[4500] (1348 bytes)
10[ENC] parsed IKE_AUTH request 1 [ EF(1/4) ]
10[ENC] received fragment #1 of 4, waiting for complete IKE message
14[NET] received packet: from 2409:8a4c:cbe:4b50:a814:e811:3751:bf8a[50995] to 2600:1900:4120:c5b7:0:3::[4500] (1348 bytes)
14[ENC] parsed IKE_AUTH request 1 [ EF(3/4) ]
14[ENC] received fragment #3 of 4, waiting for complete IKE message
09[NET] received packet: from 2409:8a4c:cbe:4b50:a814:e811:3751:bf8a[50995] to 2600:1900:4120:c5b7:0:3::[4500] (379 bytes)
09[ENC] parsed IKE_AUTH request 1 [ EF(4/4) ]
09[ENC] received fragment #4 of 4, reassembled fragmented IKE message (4236 bytes)
09[ENC] parsed IKE_AUTH request 1 [ IDi CERT N(INIT_CONTACT) CERTREQ AUTH CPRQ(ADDR ADDR6 DNS DNS6) N(ESP_TFC_PAD_N) SA TSi TSr N(MOBIKE_SUP) N(NO_ADD_ADDR) N(MULT_AUTH) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
09[IKE] received cert request for "CN=Root CA"
09[IKE] received 137 cert requests for an unknown ca
09[IKE] received end entity cert "CN=root"
09[CFG] looking for peer configs matching 2600:1900:4120:c5b7:0:3::[%any]...2409:8a4c:cbe:4b50:a814:e811:3751:bf8a[CN=root]
09[CFG] selected peer config 'ikev2-pubkey'
09[CFG]   using trusted ca certificate "CN=Root CA"
09[CFG] checking certificate status of "CN=root"
09[CFG]   using trusted certificate "CN=Root CA"
09[CFG]   crl correctly signed by "CN=Root CA"
09[CFG]   crl is valid: until Nov 08 10:18:47 2021
09[CFG]   using cached crl
09[CFG] certificate status is good
09[CFG]   reached self-signed root ca with a path length of 0
09[CFG]   using trusted certificate "CN=root"
09[IKE] authentication of 'CN=root' with RSA_EMSA_PKCS1_SHA2_256 successful
09[IKE] received ESP_TFC_PADDING_NOT_SUPPORTED, not using ESPv3 TFC padding
09[IKE] peer supports MOBIKE
09[IKE] authentication of 'wower.cool' (myself) with RSA_EMSA_PKCS1_SHA2_256 successful
09[IKE] IKE_SA ikev2-pubkey[53] established between 2600:1900:4120:c5b7:0:3::[wower.cool]...2409:8a4c:cbe:4b50:a814:e811:3751:bf8a[CN=root]
09[IKE] sending end entity cert "CN=server"
09[IKE] peer requested virtual IP %any
09[CFG] reassigning offline lease to 'CN=root'
09[IKE] assigning virtual IP 10.20.0.3 to peer 'CN=root'
09[IKE] peer requested virtual IP %any6
09[CFG] reassigning offline lease to 'CN=root'
09[IKE] assigning virtual IP fd00:bbbb:10:20::3 to peer 'CN=root'
09[CFG] selected proposal: ESP:AES_GCM_16_128/NO_EXT_SEQ
09[IKE] CHILD_SA ikev2-pubkey{50} established with SPIs cd820168_i ce83d893_o and TS 0.0.0.0/0 ::/0 === 10.20.0.3/32 fd00:bbbb:10:20::3/128
09[ENC] generating IKE_AUTH response 1 [ IDr CERT AUTH CPRP(ADDR ADDR6 DNS DNS DNS6 DNS6) SA TSi TSr N(MOBIKE_SUP) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_6_ADDR) N(ADD_6_ADDR) N(ADD_6_ADDR) ]
09[ENC] splitting IKE message (1539 bytes) into 2 fragments
09[ENC] generating IKE_AUTH response 1 [ EF(1/2) ]
09[ENC] generating IKE_AUTH response 1 [ EF(2/2) ]
09[NET] sending packet: from 2600:1900:4120:c5b7:0:3::[4500] to 2409:8a4c:cbe:4b50:a814:e811:3751:bf8a[50995] (1228 bytes)
09[NET] sending packet: from 2600:1900:4120:c5b7:0:3::[4500] to 2409:8a4c:cbe:4b50:a814:e811:3751:bf8a[50995] (376 bytes)
15[IKE] sending DPD request
15[ENC] generating INFORMATIONAL request 0 [ ]
15[NET] sending packet: from 2600:1900:4120:c5b7:0:3::[4500] to 2409:8a4c:cbe:4b50:a814:e811:3751:bf8a[50995] (57 bytes)
13[NET] received packet: from 2409:8a4c:cbe:4b50:a814:e811:3751:bf8a[50995] to 2600:1900:4120:c5b7:0:3::[4500] (57 bytes)
13[ENC] parsed INFORMATIONAL response 0 [ ]
15[IKE] sending DPD request
15[ENC] generating INFORMATIONAL request 1 [ ]
15[NET] sending packet: from 2600:1900:4120:c5b7:0:3::[4500] to 2409:8a4c:cbe:4b50:a814:e811:3751:bf8a[50995] (57 bytes)
08[NET] received packet: from 2409:8a4c:cbe:4b50:a814:e811:3751:bf8a[50995] to 2600:1900:4120:c5b7:0:3::[4500] (57 bytes)
08[ENC] parsed INFORMATIONAL response 1 [ ]

**Comment by ghost** [comment, marked as answer]
Now Android strongswan client set tunnel mtu 1360, and this problen disappeared. Linux client use "ip link set dev interface mtu 1360" can't solve the problem.This should be an MTU/MSS issue。

**Reply by Thermi** [reply]
If the lowest possible MTU is 1360, then of course setting the MSS to that value won't solve the problem.
